crates/typos-cli/src/bin/typos-cli/main.rs

```diff
@@ -156,8 +159,26 @@ fn run_checks(args: &args::Args) -> proc_exit::ExitResult {
 
     let mut typos_found = false;
     let mut errors_found = false;
-    for path in args.path.iter() {
+
+    let stdin_paths = if args.stdin_paths {
+        Some(
+            io::stdin()
+                .lines()
+                .map(|res| res.map(PathBuf::from))
+                .collect::<Result<_, _>>()
+                .with_code(proc_exit::sysexits::IO_ERR)?,
+        )
+    } else if args.stdin_paths0 {
+        Some(unix_read_paths_from_stdin()?)
+    } else {
+        None
+    };
+
+    for path in stdin_paths.as_ref().unwrap_or(&args.path) {
```

Comment:
Since the stdin paths don't conflict with user provided paths in clap, this would be ignoring user data

If you decide to make them all mutually exclusive, you can add a `group = "source"` to each argument.  By default, clap treats groups as mutually exclusive

Reply:
TIL.

Reply:
The biggest problem is the default value (`.`). If it wasn't for that we could just use `.chain` and use both stdin-paths and cli passed onces. 

Though for simplicity sake, it's probably best to forbid using both.